### PR TITLE
Allow configuring timeouts to skip slow server groups from queries

### PR DESCRIPTION
**Pull request by omegas27**

### CODE REVIEW

**Comment by jacksontj** [comment]
This feature seems like a good addition, one thing I'll clarify here is that this enforces a timeout which by default won't skip-- but combined with other options you can get the behavior you are looking for (I assume you know, mostly just odcing here for posterity :) ).

Just need to get CI to pass, I think https://github.com/jacksontj/promxy/pull/360 will fix it? But we'll have to see.

**Comment by omegas27** [comment]
I wanted to try this but I guess adding an extra round tripper around the transport one would make more sense

**Comment by omegas27** [comment]
I think I misinterpreted your earlier message. Let me revert it back

File: pkg/servergroup/config.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ type Config struct {
 	AntiAffinity time.Duration `yaml:"anti_affinity,omitempty"`
 
 	// Timeout, if non-zero, specifies the amount of
-	// time to wait for a server's response headers after fully
+	// time in seconds to wait for a server's response headers after fully
```
Comment (jacksontj) [inline pkg/servergroup/config.go:107]: We actually don't want to say in seconds as this is a `time.Duration` so it'll properly parse durations such as `1s` or `5m` -- so IMO this second commit is unnecessary (and doesn't match the other duration stuff such as anti_affinity above).

**Review by jacksontj — COMMENTED** [review]
One last comment in, I think we just drop the second commit and we're good to merge :)

File: pkg/servergroup/config.go (hunk 1 above)
Comment (omegas27) [reply pkg/servergroup/config.go:107]: Thanks for your feedback on this and help to push this forward :)

Comment (omegas27) [reply pkg/servergroup/config.go]: I rebased from upstream/master

**Comment by jacksontj** [comment]
Thanks for the contribution! :tada:

**Pull request merged**